internal/controller/nats/unit_test.go

```diff
@@ -94,3 +98,13 @@ func (testEnv *MockedUnitTestEnvironment) GetNATS(name, namespace string) (natsv
 	}, &nats)
 	return nats, err
 }
+
+func (testEnv *MockedUnitTestEnvironment) GetK8sEvents() []string {
+	eventList := make([]string, 0)
+	i := len(testEnv.Recorder.Events)
+	for j := 0; j < i; j++ {
```

Comment:
`range` isn't working

Reply:
did you try like this and it didn't work? 
```
eventList := make([]string, 0, cap(testEnv.Recorder.Events))
for _, event := range testEnv.Recorder.Events {
    eventList = append(eventList, event)
}
```
It hangs if  testEnv.Recorder.Events channel is not closed.

Reply:
I asked this because during looping new messages may come and you miss that message with the current loop with length. I this is okay for testing as it is not a productive code.